src/openscad.cc

```diff
@@ -732,6 +732,20 @@ void registerDefaultIcon(QString applicationFilePath) {
 void registerDefaultIcon(const QString&) { }
 #endif
 
+static void assertValidPath()
+{
+#ifndef __WIN32
+  boost::system::error_code ec{};
+  auto current_path = fs::current_path(ec);
+
+  while(ec.value() != 0)
+  {
+      fs::current_path(fs::path(".."));
```

Comment:
Why not `current_path.parent_path()`

Reply:
I tried it. It doesn't work. any boost::fs operation on a current path nonexistent will fail and throw an exception.